nilearn/decoding/decoder.py

```diff
@@ -411,23 +423,18 @@ def fit(self, X, y, groups=None):
         """
         # Setup memory
         self.memory_ = _check_memory(self.memory, self.verbose)
+
+        # Check for valid estimator and memory type
```

Comment:
This comment, and the one above it, are unnecessary, and contributes noise. The code itself explains what is being done here, as code should, as you chose a good name for the function.